### PR TITLE
[wasm] Do not build mono libs with `-msimd128`

### PR DESCRIPTION
**Pull request by radekdoulik**
Make it optional, move minimal set of code, which requires `-msimd128`, to separate library. Also provide "stub" nosimd version of that library.

Choose the appropriate library during app linking.

Part of solution for https://github.com/dotnet/runtime/issues/89302

### CODE REVIEW

File: src/mono/mono/mini/interp/transform-simd.c [hunk 1]
```diff
@@ -900,6 +901,9 @@ interp_emit_simd_intrinsics (TransformData *td, MonoMethod *cmethod, MonoMethodS
 	if (image != mono_get_corlib ())
 		return FALSE;
 
+	if (!interp_simd_enabled)
+		return FALSE;
+
```
Comment (radekdoulik) [inline src/mono/mono/mini/interp/transform-simd.c:906]: @kg with this I am able to build and run browser-bench sample with and without SIMD enabled. Does it look right or did I disable too much parts here?

Comment (kg) [reply src/mono/mono/mini/interp/transform-simd.c:906]: This looks right to me, and afaik the only place we explicitly use simd opcodes in the interp is in the C intrinsics which will no longer get emitted

Comment (radekdoulik) [reply src/mono/mono/mini/interp/transform-simd.c:906]: I think I will need to split the interp-simd.c to all platforms and wasm specific. the build is now broken on other platforms. this way we can probably still get some perf speedup on Vector128, which I think currently doesn't use wasm simd instructions, right?

Comment (radekdoulik) [reply src/mono/mono/mini/interp/transform-simd.c:906]: In the end I didn't split it, just fixed the other platforms and wasi builds. We can split it later if needed.

**Comment by pavelsavara** [comment]
could we have WBT for this ?

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props [hunk 2]
```diff
@@ -240,6 +240,8 @@
     <PlatformManifestFileEntry Include="libmono-profiler-browser.a" IsNative="true" />
     <PlatformManifestFileEntry Include="libmono-wasm-eh-js.a" IsNative="true" />
     <PlatformManifestFileEntry Include="libmono-wasm-eh-wasm.a" IsNative="true" />
+    <PlatformManifestFileEntry Include="libmono-wasm-simd.a" IsNative="true" />
```
[the remark this answers is not in the document]
Comment (kg) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:243]: I can't imagine a good reason to enable SIMD but disable EH.

[the remark this answers is not in the document]
Comment (radekdoulik) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.props:243]: These don't overlap, so we don't need 4 combinations.

**Comment by radekdoulik** [comment]
> could we have WBT for this ?

yeah, I plan to add tests later in another PR, with wa-info involved.

**Review by kg — APPROVED** [review]
I'm not 100% confident in this, but it looks mergeable to me. One big caveat:
The result of this is that the `EnableSIMD != 'true'` scenario will be much slower than it was before I added interp PackedSimd support. Before, we had support for all the non-packedsimd SIMD APIs in the interpreter, even if the actual implementations were not using native WebAssembly SIMD. Those APIs resulted in sizable performance improvements. Because you're now disabling *all* SIMD instead of just PackedSimd, users with EnableSIMD set to false will probably experience performance worse than net7.

AFAIK the only part of the interpreter that actually relies on msimd128 is PackedSimd.

**Comment by radical** [comment]
We should add a test to build, and run an app with simd turned off.

**Pull request merged**